drivers/include/sdmmc/sdmmc.h

```diff
@@ -0,0 +1,1632 @@
+    uint8_t  VIDEO_SPEED_CLASS : 8;         /**< [391:384] Section 4.10.2.10, Table 4-54 */
+    uint8_t  ERASE_TIMEOUT : 6;             /**< [407:402] Section 4.10.2.6, Table 4-50 */
+    uint8_t  ERASE_OFFSET : 2;              /**< [401:400] Section 4.10.2.7, Table 4-51 */
+    uint8_t  UHS_SPEED_GRADE : 4;           /**< [399:396] Section 4.10.2.8, Table 4-52 */
+    uint8_t  UHS_AU_SIZE : 4;               /**< [395:392] Section 4.10.2.9, Table 4-3 */
+    uint8_t  AU_SIZE : 4;                   /**< [431:428] Section 4.10.2.4, Table 4-47 */
+    uint8_t  DAT_BUS_WIDTH : 2;             /**< [511:510] Section 4.10.2, Table 4-44 */
+    uint8_t  SECURED_MODE : 1;              /**< [509] Section 4.10.2, Table 4-44 */
+} sdmmc_sd_status_t;
+
+/** @} */
+
+/**
+ * @name CID register (SD Memory and MMC Cards)
+ *
+ * @see Physical Layer Simplified Specification Version 9.00, Section 5.2,
+ *      Table 5.2 [[sdcard.org](https://www.sdcard.org)]
+ * @{
+ */
+
+/** CID register size in bytes (CID is 128 bit) */
+#define SDMMC_CID_REG_SIZE  (16)
+
+#define SDMMC_CID_OID_SIZE  (2)   /**< OID (OEM/Application ID) size in byte */
+#define SDMMC_CID_PNM_SIZE  (5)   /**< PNM (Product name) size in byte */
+
+/**
+ * @brief   CID register structure (SD Memory and MMC Cards)
+ *
+ * @see Physical Layer Simplified Specification Version 9.00, Section 5.2,
+ *      Table 5.2 [[sdcard.org](https://www.sdcard.org)]
+ */
+typedef struct {
+    uint8_t MID;                    /**< Manufacturer ID */
+    char OID[SDMMC_CID_OID_SIZE];   /**< OEM/Application ID*/
+    char PNM[SDMMC_CID_PNM_SIZE];   /**< Product name */
+    uint8_t PRV;                    /**< Product revision */
+    uint32_t PSN;                   /**< Product serial number */
+    uint16_t MDT;                   /**< Manufacturing date */
+    uint8_t CID_CRC;                /**< CRC7 checksum */
+} sdmmc_cid_t;
+
+/** @} */
+
+/**
+ * @name CSD Register Definitions
+ *
+ * @see Physical Layer Simplified Specification Version 9.00, Section 5.3,
+ *      [[sdcard.org](https://www.sdcard.org)] \n
+ *      JEDEC Standard No. JESD84-B42, MultiMediaCard (MMC) Electrical
+ *      Standard, High Capacity (MMCA, 4.2), Section 8.3, Table 35
+ *      [[jedec.org](https://www.jedec.org)]
+ * @{
+ */
+
+/**
+ * @brief   CSD Register Versions (SD Memory Card and MMC)
+ */
+typedef enum {
+    SDMMC_CSD_V1 = 0,   /**< CSD Version 1.0 (SDSC and MMC) */
+    SDMMC_CSD_V2 = 1,   /**< CSD Version 2.0 (SDHC/SDXC) or Version 1.1 (MMC) */
+    SDMMC_CSD_V3 = 2,   /**< CSD Version 3.0 (SDUC) or Version 1.2 (MMC) */
+    SDMMC_CSD_Vx = 3,   /**< Reserved (SD) or Version in EXT_CSD (MMC) */
+} sdmmc_csd_version_t;
+
+/** CSD register size in byte (CSD is 128 bit) */
+#define SDMMC_CSD_REG_SIZE  (16)
+
+/**
+ * @brief   CSD register structure Version 1.0
+ *
+ * @see Physical Layer Simplified Specification Version 9.00, Section 5.3.2,
+ *      Table 5.4 [[sdcard.org](https://www.sdcard.org)]
+ */
+typedef struct {
```

Comment:
```suggestion
typedef struct __attribute__((packed)) {
```

same here and below since those are read directly from memory